### PR TITLE
System api

### PR DESCRIPTION
**Pull request by Neutrollized**
**Issue:**  generator/system.go is supposed to use POST and not PUT, but [Nomad's agent/system_endpoint.go](https://github.com/hashicorp/nomad/blob/main/command/agent/system_endpoint.go) only accepts PUT.  As a result, the v1/system_test.go gets a 405 invalid method error which is absolutely valid.

**What I've tried:** changing http.MethodPost to http.MethodPut, and `make v1` builds successfully but then test will complain about v1/system.go's SystemApi 

How sure how to approach it at this point :(

### CODE REVIEW

**Comment by DerekStrickland** [comment]
I wondered if we were going to run into this!  Thx for the hard work!  I'll add support for PUT and update you so you can rebase and then this should work.  Thanks again so much for your hard work.

File: v1/system.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+func (c *Client) System() *System {
+	return &System{client: c}
+}
+
+func (s *System) SystemApi() *client.SystemApiService {
+	return s.client.apiClient.SystemApi
+}
+
+func (s *System) GarbageCollection(ctx context.Context) (interface{}, error) {
```
Comment (DerekStrickland) [inline v1/system.go:21]:
```suggestion
func (s *System) GarbageCollect(ctx context.Context) (interface{}, error) {
```

File: v1/system.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+	result, err := s.client.ExecNoResponseRequest(ctx, request)
+	if err != nil {
+		return result, err
+	}
+
+	return result, nil
+}
+
+func (s *System) Reconcile(ctx context.Context) (interface{}, error) {
```
Comment (DerekStrickland) [inline v1/system.go:32]: What is this result?  Should this just return err?

File: v1/system.go (hunk 1 above)
Comment (DerekStrickland) [reply v1/system.go:21]: Should this just return err?  What is this result?

File: v1/api.go [hunk 3]
```diff
@@ -147,9 +150,30 @@ func (c *Client) ExecRequest(_ context.Context, request interface{}) (interface{
...
+	if !ok {
+		return nil, errors.New("ExecNoResponseRequest failed: no Execute method on interface")
+	}
+
+	values := valueOf.MethodByName("Execute").Call([]reflect.Value{})
+	if !values[1].IsNil() {
+		return nil, values[1].Interface().(error)
+	}
+
+	result := values[0].Interface()
```
Comment (DerekStrickland) [inline v1/api.go:169]: I'm confused by this.  If this returns no response, why does this method return a result?  Is this just the HTTP response?  If so, I think we don't need to return it.

File: v1/system.go (hunk 1 above)
Comment (Neutrollized) [reply v1/system.go:21]: I had tried to get it to just return err, but I couldn't get the test to run with an error.  So I ended up attaching the HTTP response as a result and the test just checks to see that it's not nil.

You're absolutely right, the result has no bearing on the  test, I just couldn't get the test to compile/run.

File: v1/api.go (hunk 3 above)
Comment (Neutrollized) [reply v1/api.go:169]: I had tried to get it to just return err, but I couldn't get the test to run with an error.  So I ended up attaching the HTTP response as a result and the test just checks to see that it's not nil.

You're absolutely right, the result has no bearing on the  test, I just couldn't get the test to compile/run.

Comment (Neutrollized) [reply v1/api.go:169]: Added!

If I update the test to:
```
		err := testClient.System().GarbageCollect(writeOpts.Ctx())
		require.NoError(t, err)
```

and I get: `./system_test.go:16:7: no new variables on left side of :=`

...or am I supposed to just run the test without assigning the output to a variable?  i.e. `testClient.System().GarbageCollect(writeOpts.Ctx())`

**Pull request merged**